src/rpc/common/impl/Processors.h

```diff
@@ -44,12 +46,24 @@ struct DefaultProcessor final
                 return Error{ret.error()};  // forward Status
 
             auto const inData = value_to<typename HandlerType::Input>(value);
-
-            // real handler is given expected Input, not json
-            if (auto const ret = handler.process(inData); not ret)
-                return Error{ret.error()};  // forward Status
+            if constexpr (NonCoroutineProcess<HandlerType>)
+            {
+                auto const ret = handler.process(inData);
+                // real handler is given expected Input, not json
+                if (!ret)
+                    return Error{ret.error()};  // forward Status
+                else
+                    return value_from(ret.value());
+            }
             else
-                return value_from(ret.value());
+            {
+                auto const ret = handler.process(inData, ptrYield);
```

Comment:
Here `ptrYield` can be `nullptr`. I know this is detail code and so we should not be calling it with a `nullptr` but it's always better to make strong APIs that are hard(er) to misuse.
I'd just pass references everywhere instead and call with `yield` if the handler implements it, otherwise call without `yield`. I think it's better to rely on overload resolution in this case. If the (constexpr) logic becomes too heavy in one place you can always use private member functions to refactor to smaller bits.